src/ds/ds-options.h

```diff
@@ -193,16 +193,18 @@ namespace librealsense
     class external_sync_mode : public option
     {
     public:
-        external_sync_mode(hw_monitor& hwm, sensor_base* depth_ep = nullptr, int ver = 1); // ver = 1, for firmware 5.9.15.1 and later, INTERCAM_SYNC_MAX is 2 with master and slave mode only.
-                                                                                           // ver = 2, for firmware 5.12.4.0 and later, INTERCAM_SYNC_MAX is 258 by adding FULL SLAVE mode and genlock with trigger frequency 1 - 255.
-                                                                                           // ver = 3  for firmware 5.12.12.100 and later, INTERCAM_SYNC_MAX is 260 by adding genlock with laser on-off and Off-On two frames.
+        external_sync_mode( hw_monitor & hwm,
+                            const std::weak_ptr< sensor_base > & depth_ep = std::weak_ptr< sensor_base >(),
```

Comment:
Don't we want to build it with the shared ptr itself?

Reply:
We are storing a `weak_ptr` no need to build a `shared_ptr` and convert to weak.
Was a default `nullptr` pointer, now uses an empty `weak_ptr`